source/lac/trilinos_precondition_ml.cc

```diff
@@ -183,32 +183,38 @@ namespace TrilinosWrappers
     if (constant_modes_dimension > 0)
       {
         const size_type n_rows = n_global_rows(matrix);
-        const bool constant_modes_are_global =
-          additional_data.constant_modes[0].size() == n_rows;
-        const size_type n_relevant_rows =
-          constant_modes_are_global ? n_rows : additional_data.constant_modes[0].size();
+        const bool constant_modes_are_global
+          = additional_data.constant_modes[0].size() == n_rows;
         const size_type my_size = domain_map.NumMyElements();
-        if (constant_modes_are_global == false)
-          Assert (n_relevant_rows == my_size,
-                  ExcDimensionMismatch(n_relevant_rows, my_size));
         Assert (n_rows ==
                 static_cast<size_type>(global_length(distributed_constant_modes)),
                 ExcDimensionMismatch(n_rows,
                                      global_length(distributed_constant_modes)));
 
-        (void)n_relevant_rows;
-        (void)global_length;
```

Comment:
Ah, I see, the line then clearly makes no sense. It's not wrong -- it simply takes the address of a function and then discards the value. This has no side effect and you can indeed remove the line.